controller/src/beerocks/master/son_actions.cpp

```diff
@@ -505,3 +505,70 @@ bool son_actions::send_ap_config_renew_msg(ieee1905_1::CmduMessageTx &cmdu_tx, d
 
     return result;
 }
+
+bool son_actions::send_beacon_metrics_query_msg(ieee1905_1::CmduMessageTx &cmdu_tx, db &database,
+                                                const sMacAddr &sta_mac)
+{
+    std::string bssid     = database.get_node_parent(tlvf::mac_to_string(sta_mac));
+    std::string radio_mac = database.get_node_parent_radio(bssid);
+    auto agent_mac        = database.get_node_parent_ire(bssid);
+
+    auto measurement_request =
+        cmdu_tx.create(0, ieee1905_1::eMessageType::BEACON_METRICS_QUERY_MESSAGE);
+
+    if (!measurement_request) {
+        LOG(ERROR) << "Failed building BEACON_METRICS_QUERY_MESSAGE";
+        return false;
+    }
+
+    auto tlvBeaconMtericsQuery = cmdu_tx.addClass<wfa_map::tlvBeaconMetricsQuery>();
+
+    if (!tlvBeaconMtericsQuery) {
+        LOG(ERROR) << "Failed addClass<wfa_map::tlvBeaconMetricsQuery>";
+        return false;
+    }
+
+    tlvBeaconMtericsQuery->associated_sta_mac() = sta_mac;
+    tlvBeaconMtericsQuery->operating_class() =
+        database.get_hostap_operating_class(tlvf::mac_from_string(bssid));
+    tlvBeaconMtericsQuery->channel_number() = database.get_node_channel(bssid);
+    tlvBeaconMtericsQuery->bssid()          = tlvf::mac_from_string(bssid);
```

Comment:
I don't understand why you filled it as like we will always request measurement on the bssid that the station is connected to?
Optimal path request beacon measurements on other bssids, in other channels/operating classes.